From 78231f00b786e8da20a49da2861285ae440b0bf8 Mon Sep 17 00:00:00 2001
From: MagneFire <IDaNLContact@gmail.com>
Date: Mon, 31 Aug 2020 19:03:58 +0200
Subject: [PATCH] Use variant when setting properties.
 org.freedesktop.DBus.Properties.Set expects a variant containing a type and
 value(variant:double:0.4). Previously it would just send the type and value
 (double:0.4). This would result in the following error:   string "No such
 method 'Set' in interface 'org.freedesktop.DBus.Properties' at object path
 '/org/mpris/MediaPlayer2' (signature 'ssd')"

Upstream-Status: Inappropriate

---
 src/dbusextendedabstractinterface.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dbusextendedabstractinterface.cpp b/src/dbusextendedabstractinterface.cpp
index 935278e..9348ab8 100644
--- a/src/dbusextendedabstractinterface.cpp
+++ b/src/dbusextendedabstractinterface.cpp
@@ -243,7 +243,7 @@ void DBusExtendedAbstractInterface::internalPropSet(const char *propname, const
             return;
         }

-        asyncSetProperty(propname, QVariant(metaProperty.type(), propertyPtr));
+        asyncSetProperty(propname, QVariant::fromValue(QDBusVariant(QVariant(metaProperty.type(), propertyPtr))));
     }
 }

--
2.28.0
